src/ChatClient.ts

```diff
@@ -144,23 +144,32 @@ export class ChatClient {
   }
 
   async join() {
-    const {email, channelId, teamId, channelName} = await this.sendCommand<any, any>('join', {
-      version: getAppVersion(),
-    });
-    if (!email || !channelId || !teamId || !channelName) {
-      console.error(`Client ready without email or channel. Disconnecting...`);
-      this.client.close();
-    }
+    try {
+      const {email, channelId, teamId, channelName} = await this.sendCommand<any, any>('join', {
+        version: getAppVersion(),
+      });
+      if (!email || !channelId || !teamId || !channelName) {
+        console.error(`Client ready without email or channel. Disconnecting...`);
+        this.client.close();
+        this.fireError(new Error('No mapping found.'));
+        return;
+      }
 
-    this.email = email;
-    this.teamId = teamId;
-    this.channelId = channelId;
-    this.channelName = channelName;
+      this.email = email;
+      this.teamId = teamId;
+      this.channelId = channelId;
+      this.channelName = channelName;
 
-    console.log(`Client ready: ${email}, ${channelId}`);
-    this.fireStatusChange(ConnectionStatus.CONNECTED);
+      console.log(`Client ready: ${email}, ${channelId}`);
+      this.fireStatusChange(ConnectionStatus.CONNECTED);
 
-    await this.requestHistory();
+      await this.requestHistory();
+    } catch (e: any) {
+      console.error(`Impossible to join. Disconnecting...`);
```

Comment:
When we catch the error here, we just close the socket once again, so I am trying to understand why the fix actually works?